File: scala-pr/54/scala/code-quality-error-handling/SafeJsonParser.scala
import scala.util.parsing.json.JSON
import scala.util.{Try, Success, Failure}

class SafeJsonParser {
  def parseJson(jsonString: String): Try[Any] = {
    // {fact rule=code-quality-error-handling@v1.0 defects=0}
    if (jsonString == null || jsonString.isEmpty) {
      return Failure(new IllegalArgumentException("JSON string cannot be null or empty"))
    }

    Try {
      JSON.parseFull(jsonString) match {
        case Some(result) => result
        case None => throw new IllegalArgumentException("Invalid JSON format")
      }
    }.recoverWith {
      case ex: IllegalArgumentException => Failure(ex)
      case ex => Failure(new RuntimeException(s"Unexpected error parsing JSON: ${ex.getMessage}", ex))
    }
    // {/fact}
  }
}

File: kotlin-pr/50/kotlin/code-quality-error-handling/SafeJsonParser.kt
import kotlinx.serialization.json.Json
import kotlinx.serialization.SerializationException

class SafeJsonParser {
    fun parseJson(jsonString: String): Result<Any> {
        // {fact rule=code-quality-error-handling@v1.0 defects=0}
        if (jsonString.isBlank()) {
            return Result.failure(IllegalArgumentException("JSON string cannot be null or empty"))
        }

        return try {
            val result = Json.parseToJsonElement(jsonString)
            Result.success(result)
        } catch (ex: SerializationException) {
            Result.failure(IllegalArgumentException("Invalid JSON format: ${ex.message}", ex))
        } catch (ex: Exception) {
            Result.failure(RuntimeException("Unexpected error parsing JSON: ${ex.message}", ex))
        }
        // {/fact}
    }
}